app/models/gi_bill_feedback.rb

```diff
@@ -20,6 +34,50 @@ def get_user_details
     { 'profile_data' => profile_data }
   end
 
+  def remove_malformed_options(options_hash)
+    return_val = {}
+    options_hash.each do |k, v|
+      next if FEEDBACK_MAPPINGS.keys.include?(k)
+
+      return_val[k] = v
+    end
+
+    return_val
+  end
+
+  def transform_malformed_options(options_hash)
+    return_val = {}
+
+    options_hash.each do |k, v|
+      FEEDBACK_MAPPINGS[k].tap do |new_key|
```

Comment:
Is there a reason to use `tap` here instead of just `if FEEDBACK_MAPPINGS[k].blank?`? Also, `map` might be a tidier option than `each`.

Reply:
i'm using tap because the value of `FEEDBACK_MAPPINGS[k]` is used twice

map would transform the hash into an array and i don't want that

Reply:
Makes sense on both counts! Thanks for the clarification